bluemira/base/parameter.py

```diff
@@ -94,6 +94,27 @@ def __str__(self):
         """
         return repr(self.to_dict())
 
+    def __setattr__(self, attr: str, value: Union[bool, str]):
+        """
+        Protect against additional attributes
```

Comment:
dataclass with __slots__ comes in 3.10 :'(

Reply:
...yeah very annoying, but at least its coming